chipsec/chipset.py

```diff
@@ -855,6 +865,15 @@ def read_register(self, reg_name, cpu_thread=0, bus_index=0):
 
         return reg_value
 
+    def read_register_all(self, reg_name, cpu_thread=0):
+        values = []
+        bus_data = self.get_register_bus( reg_name )
+        if bus_data is None:
+            return None
+        for index in range( len(bus_data) ):
```

Comment:
get_register_bus returns a list of the buses, should just loop through those instances.  The instances are returned in string, so you'll need to convert them to integer.

Reply:
@frinzell please let me know why range is being used instead of the list of bus entries

Reply:
`index` in `self.read_register()`  is index (0,1,2...) of the bus list not the actual bus number (0x1D, 0x1F,..)